### PR TITLE
 Add openHAB Super POM

### PR DESCRIPTION
**Pull request by pfink**
@kaikreuzer @maggu2810 Feel free to review!

Signed-off-by: Patrick Fink <mail@pfink.de>

### CODE REVIEW

**Review by maggu2810 — COMMENTED** [review]
If your magic Super POM should be used for all openHAB POM files as base I would prefer to prefix pure openHAB properties by "oh" or "oh." or something similar.
As more and more projects are "controlled" by this Super POM you should not rely that no one (e.g. someone that is using the addons POM as its parent for non openHAB maintained addons) already uses properties like the four you used here.

WDYT about `oh.scm.gitBaseUrl`, `oh.repo....` etc.?

**Comment by pfink** [comment]
Done!

**Review by kaikreuzer — APPROVED** [review]
@maggu2810 Are you ok with this super pom now?
If so, I will merge, as it is a crucial step towards fixing our release build.

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+    <repository>
+      <releases>
+        <enabled>true</enabled>
+        <updatePolicy>never</updatePolicy>
+      </releases>
+      <snapshots>
+        <enabled>false</enabled>
+      </snapshots>
+      <id>openhab-artifactory-release</id>
```
Comment (maggu2810) [inline pom.xml]: As "${oh.repo.releaseBaseUrl}" is used I wonder about the "artifactory" in the ID.
Shouldn't the ID be e.g. `openhab-release` as artifactory is just a repository manager and does not need to be used as you allow the change the URL to another one.

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      <name>JFrog Artifactory Repository</name>
```
Comment (maggu2810) [inline pom.xml]: Perhaps (WRT the comment about the ID) "openHAB Release Repository" would be a better name.

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  </repositories>
+
+  <pluginRepositories>
+    <pluginRepository>
+      <id>jcenter</id>
+      <name>JCenter Repository</name>
+      <url>https://jcenter.bintray.com</url>
+    </pluginRepository>
+    <pluginRepository>
+      <id>openhab-artifactory-release</id>
```
Comment (maggu2810) [inline pom.xml]: see above

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+      <repositories>
+        <repository>
+          <releases>
+            <enabled>false</enabled>
+          </releases>
+          <snapshots>
+            <enabled>true</enabled>
+            <updatePolicy>daily</updatePolicy>
+          </snapshots>
+          <id>openhab-artifactory-snapshot</id>
```
Comment (maggu2810) [inline pom.xml]: name + id: see above

File: pom.xml [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+              <artifactId>unleash-scm-provider-git</artifactId>
+              <version>2.2.0</version>
+            </dependency>
+          </dependencies>
+        </plugin>
+      </plugins>
+    </pluginManagement>
+  </build>
+
+</project>
```
Comment (maggu2810) [inline pom.xml]: Can you add a new line to the end?

**Review by maggu2810 — COMMENTED** [review]
minor ones

**Comment by pfink** [comment]
Thanks for review, changes done!

**Review by maggu2810 — APPROVED** [review]

File: pom.xml (hunk 5 above)
Comment (maggu2810) [reply pom.xml]: FYI It is defined by POSIX that a line ends with a newline character: https://pubs.opengroup.org/onlinepubs/9699919799.2018edition/basedefs/V1_chap03.html#tag_03_206

**Comment by maggu2810** [comment]
Thanks for changes, review done!

**Comment by kaikreuzer** [comment]
```
Failed to transfer Could not find artifact org.openhab:openhab-super-pom:pom:[1.0, 2.0) in central (https://ci.openhab.org/artifactory/libs-release)
```
@pfink How shall we actually deploy this super pom to our repo? Simply manually? Or shall we create a Jenkins job for it?

**Comment by pfink** [comment]
I'll create a Jenkins job :)

**Comment by pfink** [comment]
I opened #7 for further discussion regarding deployment.

**Pull request merged**